### DISCUSSION TITLE
No traceback for un-raised exceptions

### DISCUSSION THREAD

**Discussion by GarrisonD** [getsentry/sentry-python#2159] [Q&A]
After executing this code I have tracebacks in Sentry:

```python
try:
  raise Exception
except:
  capture_exception()
```

But after executing this, I don't:

```python
capture_exception(Exception())
```

Here is what [the docs](https://docs.sentry.io/platforms/python/configuration/options/#attach-stacktrace) say:

> Stack traces are always attached to exceptions

**Comment by GarrisonD** [comment]
Maybe a more realistic example helps:

```python
def get_something():
  something = Model.objects.filter(this='that').first() # returns None or Model but never raises exceptions

  if not something:
    capture_exception(Model.DoesNotExist())

  return something
```

I can do something like:

```python
try:
  raise Model.DoesNotExist
except:
  capture_exception()
```

but it's too much code for such a simple action...

**Comment by sl0thentr0py** [comment]
@GarrisonD if you manually instantiate exceptions this way, they don't have the `__traceback__` attribute so we have no way of knowing what that is in the SDK. You would need to somehow fetch the traceback and pass it into [`with_traceback`](https://docs.python.org/3/library/exceptions.html#BaseException.with_traceback).
